### PR TITLE
Remove core/post-comments-form block styles that are overriding button element styles

### LINKED ISSUE #42052: Post comment form block styles are overriding button elements theme.json styles
### Description

It seems like post comment form block styles are overriding button elements theme.json.

### Step-by-step reproduction instructions

- Add `theme.json` settings for core/button block:
```
	"styles": {
              "elements": {
                      "button": {
                              "border": {
	                              "color": "black",
	                              "style": "solid",
	                              "width": "1px"
                              }
              },
	...
	...
```
- Check the rendered post comments form and see if the styles are being overridden.

### Screenshots, screen recording, code snippet

image

### Environment info

- Latest Gutenberg trunk version
- WP 6.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?

Remove core/post-comments-form block styles that are overriding button element styles from `theme.json`

## Why?

These styles are preventing that the `theme.json` styles for button elements apply over the core/post-comments-form submit button.

## How?

I think those styles can be removed because they are no longer needed and they are conflicting with the button element styles.

## Testing Instructions

1. Add theme.json settings for core/button block:
```
	"styles": {
              "elements": {
                      "button": {
                              "border": {
	                              "color": "black",
	                              "style": "solid",
	                              "width": "3px"
                              }
              },
	...
```
2. Check the rendered post comments form and see if the styles are being overridden.

## Screenshots or screencast 

Using the `theme.json` from the top :arrow_up:  example:

**Before:**
image

**After:**
image

### CODE REVIEW

**Review by draganescu — COMMENTED** [review]
I think we should only remove the styles that are provided by core `theme.json`. For example `cursor:pointer` is not, so it should remain. Testing this I saw no difference because the class `.wp-block-button__link` in the button block's CSS adds `cursor:pointer`, but IMO it should not be removed - same to all other CSS that are not in `theme.json`.

**Comment by matiasbenedetto** [comment]
I updated the testing instructions following the [issue conversation](https://github.com/WordPress/gutenberg/issues/42052). 
And, as @draganescu suggested, I restored all the properties but the `border` one, which can be provided by `themes.json`.

**Review by draganescu — APPROVED** [review]
Let's land this 👍🏻

**Comment by carolinan** [comment]
Using Empty theme.
Screenshots from the block editor, not site editor.

**Before:**
WordPress 6.0, for comparison. Button styles in the editor and front do not match (Gutenberg is deactivated here).
comment-form-button-gb-decativated

With Gutenberg trunk, and changes copied from PR description and added to theme.json:
comment-form-button-gb-trunk

**After, with PR applied:**
Changes copied from PR description and added to theme.json:
comment-form-button-pr-json-changes

Without button styles in theme.json, the button has the browser default border for buttons (and I assume this is the style that will be the default style used in classic themes).

comment-form-button-has-default-border


----

There is also an issue with the border radius, but I do not know which of the button element PR's that caused it. I suspect that this might even bleed through from the WordPress admin area styles, from the .submit CSS class?

**Comment by carolinan** [comment]
I am not fond of the browser default border that will be used if theme authors do not style this button, but I also do not have a solution for it.

**Review by draganescu — DISMISSED** [review]
Let's wait on figuring out what's up with the roundness in @carolinan 's tests

**Comment by carolinan** [comment]
The `.submit` style has a higher specificity.
I assume the submit class was added at some point to match the original comment_form() function.

image

**Comment by matiasbenedetto** [comment]
The problems around border-radius found by @carolinan preexist this PR and they are not strictly related to the original intention of this PR. 
image
Despite that, I introduced a commit removing the `.submit` class introduced [in this PR](https://github.com/WordPress/gutenberg/pull/41251). Removing that class doesn't seem to be affecting the form button and fixes the `border-radius` problem which caused the weird button shape :point_up: .

Apart from that, I added a default setting in Gutenberg's default `theme.json` to remove the user-agent default button border. In this way, we are able to style the `core/post-comments-form` submit button using the theme's `theme.json` element button (`styles.elements.button`).

In this screenshot, you can see that both problems seem to be solved. For this example, I'm using `emptytheme` and these settings in the theme's `theme.json`.

```
	"styles": {
		"elements": {
			"button": {
				"border": {
					"color": "red",
					"style": "solid",
					"width": "10px"
				},
				"color": {
					"background": "orange",
					"text": "blue"
				}
			}
		}
	}	
```

image


With Gutenberg plugin disabled still are some visual discrepancies between the editor and the frontend caused by some admin CSS definitions. I may have a fix for that, I'm testing it. Due it is a core-problem that needs to be fixed in core repo I think we can move forward with this PR. 

Thanks @carolinan and @draganescu for your feedback!

**Comment by scruffian** [comment]
> removing the .submit class introduced https://github.com/WordPress/gutenberg/pull/41251

I think that class pre-dates that PR...

File: packages/block-library/src/post-comments-form/form.js [hunk 1]
```diff
@@ -34,7 +34,6 @@ const CommentsForm = () => {
 						name="submit"
 						type="submit"
 						className={ classnames(
-							'submit',
```
Comment (draganescu) [inline packages/block-library/src/post-comments-form/form.js:37]: Where else is this class? If it's not needed should we remove the directives as well?

Comment (matiasbenedetto) [reply packages/block-library/src/post-comments-form/form.js:37]: I haven't found the class anywhere else in Gutenberg codebase.
The CSS directives live in the WordPress core repo: https://github.com/WordPress/wordpress-develop/blob/0a17a80bccd452b91c3b63f71f010ba131a2c954/src/wp-admin/css/forms.css#L415-L421

**Review by michalczaplinski — APPROVED** [review]
This looks good to me 👍 

I have originally added the `submit` classes in 08d34f36ff2ebd35bb5dbdac63f26be5b76c8fb6 when creating the Post Comments Form for the editor. The only reason for adding it was because the PHP (non-block) comments also include it.

**Review by michalczaplinski — CHANGES_REQUESTED** [review]
Uh, I'm sorry for having to revoke my approval 😅 

I have noticed that the `submit` class is also applied on the frontend through the `comment_form_defaults` filter: https://github.com/WordPress/gutenberg/blob/c8b7177d4a672a3e0de3f7f2232f15ed4e15d78b/packages/block-library/src/post-comments-form/index.php#L32

[This](https://developer.wordpress.org/reference/functions/comment_form/#source) is where the `comment_form_defaults` filter is being applied in Core.

This filter was introduced to the Post Comments Form block in https://github.com/WordPress/gutenberg/pull/40628 because of the feedback in https://github.com/WordPress/gutenberg/issues/34994#issuecomment-1109754877.

So, I think we should either remove the `submit` class in both places or keep it in both 🙂 .

**Comment by matiasbenedetto** [comment]
Thanks for your detailed feedback @michalczaplinski !
In [the last commit](https://github.com/WordPress/gutenberg/pull/42053/commits/4912608ee39daeb16739448746f907e61b94e5ed), I removed the `submit` class also from the server-side rendering.  Do you think that could work?

**Review by michalczaplinski — APPROVED** [review]
I think that's fine! 🙂 

There is only one class (`submit`) applied to the button which is defined [here](https://github.com/WordPress/wordpress-develop/blob/3a518382cb455756b81961bfb558f0e7a012eee7/src/wp-includes/comment-template.php#L2491) and added to the final markup [here](https://github.com/WordPress/wordpress-develop/blob/3a518382cb455756b81961bfb558f0e7a012eee7/src/wp-includes/comment-template.php#L2693)

Also, I wanna note [this comment](https://github.com/WordPress/gutenberg/discussions/35470#discussioncomment-1450059) which confirms that CSS classes can change from time to time given good reasons (I'm paraphrasing).

**Pull request merged**